web/blog/mobilizing-zeek-logs/index.md

```diff
@@ -0,0 +1,232 @@
+
+```bash
+zeek -r trace.pcap LogAscii::use_json=T
+```
+
+As with TSV, this generates one file per log type containing the NDJSON records.
+Here are the same three entries from above in NDJSON:
+
+```json
+{"ts":1258531221.486539,"uid":"C8b0xF1gjm7rOZXemg","id.orig_h":"192.168.1.102","id.orig_p":68,"id.resp_h":"192.168.1.1","id.resp_p":67,"proto":"udp","service":"dhcp","duration":0.1638200283050537,"orig_bytes":301,"resp_bytes":300,"conn_state":"SF","missed_bytes":0,"history":"Dd","orig_pkts":1,"orig_ip_bytes":329,"resp_pkts":1,"resp_ip_bytes":328}
+{"ts":1258531680.237254,"uid":"CMsxKW3uTZ3tSLsN0g","id.orig_h":"192.168.1.103","id.orig_p":137,"id.resp_h":"192.168.1.255","id.resp_p":137,"proto":"udp","service":"dns","duration":3.780125141143799,"orig_bytes":350,"resp_bytes":0,"conn_state":"S0","missed_bytes":0,"history":"D","orig_pkts":7,"orig_ip_bytes":546,"resp_pkts":0,"resp_ip_bytes":0}
+{"ts":1258531693.816224,"uid":"CtTLWk44k1eOCdSjBf","id.orig_h":"192.168.1.102","id.orig_p":137,"id.resp_h":"192.168.1.255","id.resp_p":137,"proto":"udp","service":"dns","duration":3.7486469745635986,"orig_bytes":350,"resp_bytes":0,"conn_state":"S0","missed_bytes":0,"history":"D","orig_pkts":7,"orig_ip_bytes":546,"resp_pkts":0,"resp_ip_bytes":0}
+```
+
+Use the regular [`json`](/next/formats/json) parser to get the data flowing:
+
+```bash
+cat conn.log | tenzir 'read json --schema "zeek.conn" | taste 1'
+```
+
+The option `--schema` of the `json` reader passes a name of a known schema to
+validate the input against.
+
+:::note Zeek Schemas
+Tenzir comes with schemas for a stock Zeek. We have planned an operator that
+makes it also possible to simple rename the schema, akin to
+[`rename`](/next/operators/transformations/rename) for fields.
+:::
+
+### Streaming JSON
+
+The above one-file-per-log format is not conducive to stream processing because
+a critical piece of information is missing: the type of the log (or *schema*),
+which is only contained in the file name. So you can't just ship the data away
+and infer the type later at ease. And carrying the filename around in a side
+channel is cumbersome. Enter [JSON streaming
+logs](https://github.com/corelight/json-streaming-logs). This package adds two
+new fields: `_path` with the log type and `_write_ts` with the timestamp when
+the log was written. For example, the old `http.log` is now has an additional
+field `{"_path": "http" , ...}`. This makes it a lot easier to consume, because
+you can now concatenate the entire output and multiplex it over a single stream.
+
+This format doesn't come with stock Zeek. Use Zeek's package manager `zkg` to
+install it:
+
+```bash
+zkg install json-streaming-logs
+```
+
+Then pass the package name to the list of scripts on the command line:
+
+```bash
+zeek -r trace.pcap json-streaming-logs
+```
+
+And now you get JSON logs in the current directory. Here's the same `conn.log`
+example from above, this time as NDJSON:
```

Comment:
It's NDJSON above as well. See my main review comment for  how to better expose the unique feature.

Reply:
Yes, I'll address that.